### PR TITLE
tests: fix error conditions to be used properly.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix error conditions to be used properly.

## Motivation and Context
Code review and cleanup. 

## How Has This Been Tested?
Manually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/xl-v1-object.go
Comment (vadmeste) [inline cmd/xl-v1-object.go:683]: I don't encourage removing the update of onlineDisks here and in other places since we can forget this stuff when someone introduces new code after this line, it is better if we keep onlineDisks always updated.

Comment (harshavardhana) [reply cmd/xl-v1-object.go:683]: If we are not going to use it further its not useful, a function signature already communicates what it does.  Is onlineDisks being used again ? if not we shouldn't set it.  We should review when such a code is added in future to use onlineDisks by using the value set by the renameObject() 

Otherwise it doesn't make sense to set it IMO if we are not going to use it further.

**Comment by harshavardhana** [comment]
@fwessels any reviews?

**Review by fwessels — APPROVED** [review]
LGTM

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by deekoder** [comment]
Codecoverage should run. Blocked on merging.

**Pull request merged**
